addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/internal/Util.java

```diff
@@ -0,0 +1,71 @@
+package org.openhab.binding.hs110.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Util {
+    private static Logger log = LoggerFactory.getLogger(Util.class);
+
+    public static String decrypt(InputStream inputStream, boolean broadcast) throws IOException {
+
+        int in;
+        int key = 0x2B;
+        int nextKey;
+        StringBuilder sb = new StringBuilder();
+        while ((in = inputStream.read()) != -1) {
+
+            nextKey = in;
+            in = in ^ key;
+            key = nextKey;
+            sb.append((char) in);
+        }
+        System.out.println(sb.length());
+        log.info("{}", sb.length());
+        if (broadcast) {
+            return "{" + sb.toString().substring(1, sb.length() - 1) + "}";
+        } else {
+            return "{" + sb.toString().substring(5, sb.length() - 1) + "}";
+        }
+    }
+
+    public static int[] encrypt(String command) {
+
+        int[] buffer = new int[command.length()];
+        int key = 0xAB;
+        for (int i = 0; i < command.length(); i++) {
+
+            buffer[i] = command.charAt(i) ^ key;
+            key = buffer[i];
+        }
+        return buffer;
+    }
+
+    public static byte[] encryptWithHeader(String command) {
+
+        int[] data = encrypt(command);
+        byte[] bufferHeader = ByteBuffer.allocate(4).putInt(command.length()).array();
```

Comment:
This should be the only code remaining from 
https://github.com/Insxnity/hs-100/blob/master/src/net/insxnity/hs100/HS100.java
and/or https://gist.github.com/agent4788/81beb25cdcdbf7e9371361ca87d3b04a

I did not see their license - how do we work with that?